superset-frontend/src/dashboard/components/menu/ShareMenuItems/index.tsx

```diff
@@ -57,48 +53,29 @@ const ShareMenuItems = (props: ShareMenuItemProps) => {
     ...rest
   } = props;
 
-  const tabId = useTabId();
-
-  const getShortUrl = useUrlShortener(url || '');
-
-  async function getCopyUrl() {
-    const risonObj = getUrlParam(URL_PARAMS.nativeFilters);
-    if (typeof risonObj === 'object' || !dashboardId) return null;
-    const prevData = await getFilterValue(
-      dashboardId,
-      getUrlParam(URL_PARAMS.nativeFiltersKey),
-    );
-    const newDataMaskKey = await createFilterKey(
-      dashboardId,
-      JSON.stringify(prevData),
-      tabId,
-    );
-    const newUrl = new URL(`${window.location.origin}${url}`);
-    newUrl.searchParams.set(URL_PARAMS.nativeFilters.name, newDataMaskKey);
-    return `${newUrl.pathname}${newUrl.search}`;
-  }
-
   async function generateUrl() {
+    // chart
```

Comment:
Maybe we should have 2 separate functions for charts and dashboards?

Reply:
I agree, I was slightly confused by this but for some reason I didn't do anything about it 😆 I'll break it up 👍

Reply:
Upon closer inspection it almost feels like we should refactor the whole component to make this functionality more generic. So deferring this to a follow-up